src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -154,6 +154,136 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
             }
         }
 
+        /// <summary>
+        /// Adds the given <paramref name="classValue"/> to the <paramref name="tagHelperOutput"/>'s
+        /// <see cref="TagHelperOutput.Attributes"/>.
+        /// </summary>
+        /// <param name="tagHelperOutput">The <see cref="TagHelperOutput"/> this method extends.</param>
+        /// <param name="classValue">The class value to add.</param>
+        public static void AddClass(
+            this TagHelperOutput tagHelperOutput,
+            string classValue)
+        {
+            if (tagHelperOutput == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperOutput));
+            }
+
+            if (classValue.Contains(" "))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotContainWhitespace, nameof(classValue));
+            }
+
+            if (!tagHelperOutput.Attributes.TryGetAttribute("class", out TagHelperAttribute classAttribute))
+            {
+                tagHelperOutput.Attributes.Add("class", classValue);
+            }
+            else
+            {
+                var currentClassValue = ExtractClassValue(classAttribute);
+                char[] spaceChars = { '\u0020', '\u0009', '\u000A', '\u000C', '\u000D' };
+
+                if (spaceChars.Any(c => currentClassValue.Contains(c)) || currentClassValue.Equals(classValue))
+                {
+                    var arrayOfClasses = currentClassValue.Split(spaceChars);
+                    if (arrayOfClasses.Contains(classValue))
+                    {
+                        return;
+                    }
+                }
+
+                var newClassAttribute = new TagHelperAttribute(
```

Comment:
Surfacing a still-valid comment in old commits:
> Best way to make comparisons work everywhere is to pass the current `HtmlEncoder` into this method and down to `ExtractClassValue(...)`. Will also need to encode `classValue` before comparing it to anything and use an `HtmlString` here e.g. new `HtmlString($"{currentClassValue} {classValue}")`. Won't need `ClassAttributeHtmlContent` at all if you fix the comparisons.

I'd add that the `string.IsNullOrEmpty(currentClassValue)` case should be handled before other checks above.

Reply:
Thanks